File: J2EE_DoANMonHoc_WebTuyenDungViecLam/frontend/src/services/reviewService.ts
import axiosInstance from './axiosInstance';
import type { ApiResponse, Review, ReviewRequest, EmployerRating } from '@/types';

export const reviewService = {
  create: (data: ReviewRequest) =>
    axiosInstance.post<ApiResponse<Review>>('/reviews', data),

  checkReviewed: (jobId: number) =>
    axiosInstance.get<ApiResponse<boolean>>(`/reviews/check/${jobId}`),

  getByEmployer: (employerId: number, page = 0, size = 10) =>
    axiosInstance.get<ApiResponse<Review[]>>(`/reviews/employer/${employerId}`, {
      params: { page, size },
    }),

  getEmployerRating: (employerId: number) =>
    axiosInstance.get<ApiResponse<EmployerRating>>(`/reviews/employer/${employerId}/rating`),
};

File: J2EE_DoANMonHoc_WebTuyenDungViecLam/src/main/java/com/example/DA_WebTuyenDungViecLam/service/ReviewService.java
package com.example.DA_WebTuyenDungViecLam.service;

import com.example.DA_WebTuyenDungViecLam.dto.request.ReviewRequest;
import com.example.DA_WebTuyenDungViecLam.dto.response.EmployerRatingResponse;
import com.example.DA_WebTuyenDungViecLam.dto.response.ReviewResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface ReviewService {

    ReviewResponse create(ReviewRequest request, Long candidateId);

    Page<ReviewResponse> getByEmployer(Long employerId, Pageable pageable);

    EmployerRatingResponse getEmployerRating(Long employerId);

    boolean hasReviewed(Long candidateId, Long jobId);
}
